llvm/lib/Target/AMDGPU/SMInstructions.td

```diff
@@ -867,13 +867,104 @@ def SMRDBufferImm   : ComplexPattern<iPTR, 1, "SelectSMRDBufferImm">;
 def SMRDBufferImm32 : ComplexPattern<iPTR, 1, "SelectSMRDBufferImm32">;
 def SMRDBufferSgprImm : ComplexPattern<iPTR, 2, "SelectSMRDBufferSgprImm">;
 
+class SMRDAlignedLoadPat<PatFrag Op> : PatFrag <(ops node:$ptr), (Op node:$ptr), [{
+  // Returns true if it is a naturally aligned multi-dword load.
+  LoadSDNode *Ld = cast<LoadSDNode>(N);
+  unsigned Size = Ld->getMemoryVT().getStoreSize();
+  return (Size <= 4) || (Ld->getAlign().value() >= PowerOf2Ceil(Size));
```

Comment:
Shouldn't need the PowerOf2Ceil?

Reply:
For the DWORDX3 case. Even though the size is 12 Bytes, its natural alignment is 16 Bytes.

Reply:
Right but the PowerOf2Ceil makes no difference. Either you test 16>=12 or 16>=16, the result it the same. Also you don't need most of the parens on this line.

Reply:
Isn't it 12 >= 16 or 16 >=16?
The PowerOf2Ceil intends to catch the first case where the specified Align is smaller than its natural alignment.

Reply:
`Ld->getAlign().value()` will never be 12. There's no such thing as a non-power-of-two alignment.